addons/mail/models/ir_attachment.py

```diff
@@ -51,21 +55,22 @@ def _bus_notification_target(self):
         return self.env.user.partner_id
 
     def _attachment_format(self):
+        return [attachment._prepare_attachment_format(attachment) for attachment in self]
+
+    def _prepare_attachment_format(self, attachment):
         safari = request and request.httprequest.user_agent and request.httprequest.user_agent.browser == 'safari'
-        return [
-            {
-                'checksum': attachment.checksum,
-                'id': attachment.id,
-                'filename': attachment.name,
-                'name': attachment.name,
-                'mimetype': 'application/octet-stream' if safari and attachment.mimetype and 'video' in attachment.mimetype else attachment.mimetype,
-                'originThread': [('insert', {
-                    'id': attachment.res_id,
-                    'model': attachment.res_model,
-                })],
-            }
-            for attachment in self
-        ]
+        return {
+            'checksum': attachment.checksum,
+            'id': attachment.id,
+            'filename': attachment.name,
+            'name': attachment.name,
+            "size": attachment.file_size,
+            'mimetype': 'application/octet-stream' if safari and attachment.mimetype and 'video' in attachment.mimetype else attachment.mimetype,
```

Comment:
Maybe not in the scope of this PR but is it still necessary to alter the mimetype for safari? I think that it had to do with the fact that safari makes byte-range requests based on the mimetype and that it was not supported. But https://github.com/odoo/odoo/pull/97422 seems to indicate that it is now.